presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -384,6 +386,28 @@ private PlanWithProperties splitAggregation(AggregationNode node, PlanWithProper
         }
 
         @Override
+        public PlanWithProperties visitGroupId(GroupIdNode node, Context context)
+        {
+            PreferredProperties childPreference = context.getPreferredProperties().translate(removeGroupIdModifiedSymbols(node));
+            PlanWithProperties child = planChild(node, context.withPreferredProperties(childPreference));
+            return rebaseAndDeriveProperties(node, child);
+        }
+
+        private Function<Symbol, Optional<Symbol>> removeGroupIdModifiedSymbols(GroupIdNode node)
+        {
+            Set<Symbol> nulledSymbols = new HashSet<>(node.getDistinctGroupingColumns());
```

Comment:
not sure if this is better, but would it be more clear to construct the set of allowed symbols in one set?
like:
node.getSource().getOutputSymbols() - nulledSymbols (or some simplified form of this)